lib/utility.js

```diff
@@ -13,6 +15,27 @@ const sharp = require('../build/Release/sharp.node');
  */
 const format = sharp.format();
 
+/**
+ * An Object containing the available interpolators and their proper values
+ * @readonly
+ * @enum {string}
+ */
+const interpolators = {
+  /** [Nearest neighbour interpolation](http://en.wikipedia.org/wiki/Nearest-neighbor_interpolation). Suitable for image enlargement only. */
+  nearest: 'nearest',
+  /** [Bilinear interpolation](http://en.wikipedia.org/wiki/Bilinear_interpolation). Faster than bicubic but with less smooth results. */
+  bilinear: 'bilinear',
+  /** [Bicubic interpolation](http://en.wikipedia.org/wiki/Bicubic_interpolation) (the default). */
+  bicubic: 'bicubic',
+  /** [LBB interpolation](https://github.com/jcupitt/libvips/blob/master/libvips/resample/lbb.cpp#L100). Prevents some "[acutance](http://en.wikipedia.org/wiki/Acutance)" but typically reduces performance by a factor of 2. */
+  locallyBoundedBicubic: 'lbb',
+  /** [Nohalo interpolation](http://eprints.soton.ac.uk/268086/). Prevents acutance but typically reduces performance by a factor of 3. */
+  nohalo: 'nohalo',
+  /** [VSQBS interpolation](https://github.com/jcupitt/libvips/blob/master/libvips/resample/vsqbs.cpp#L48). Prevents "staircasing" when enlarging. */
+  vertexSplitQuadraticBasisSpline: 'vsqbs'
+};
+assert.deepStrictEqual(interpolators, enums.interpolators);
```

Comment:
Yes, I have scratched my head around this one for some time. Firstly, I decided to put the enum on a separate file so it can be accessed from `operators.js` (to validate user input) and `constructor.js` (to provide a default value), as the `utility.js` functions are decorating the Sharp class and this can't be accessed from those. Then I decided to (yes, it was painful to post the PR with that), duplicate the object because of JSDoc: the documentation wouldn't be generated for `interpolators` if I just reference the the property declared in `enums.js`. And my solution would be either to duplicate code or to hard-code the JSDoc comments with the enumerators, which could lead to a non-consitent documentation if this enum ends up being modified. I thought that if we take the first approach, at least a run-time error would be thrown.

One solution I found is to just not validate users input from `operators.js` (users should be using the enumerator anyway), this way we dont need to access the enum from there, and to hard-code a default value in `contructors.js`. I will be pushing a commit with these changes. Apart from that I can’t think of a solution other than refactoring how the Sharp class is built. Which I can for sure try to do if no other optimal solution is found. Can you give me a little more feedback on this? And sorry for all this wall of text for such a small thing. Thanks for your time.

Reply:
You should be able to access the `Sharp.interpolators` object via `this.constructor.interpolators` within the `affine()` function.